### PR TITLE
use MOCKPASS_NRIC as default if SHOW_LOGIN_PAGE is false

### PR DESCRIPTION
**Pull request by yahudev**

### CODE REVIEW

File: lib/express/myinfo.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ const MOCKPASS_PUBLIC_KEY = fs.readFileSync(path.resolve(__dirname, '../../stati
 
 const authorizations = {}
 
-function config (app, { serviceProvider, port }) {
+function config(app, { serviceProvider, port }) {
```
Comment (LoneRifle) [inline lib/express/myinfo.js:24]: Nitpick - Keep the space for now please - it's required by our linters

File: lib/express/myinfo.js [hunk 2]
```diff
@@ -114,6 +114,8 @@ function config (app, { serviceProvider, port }) {
     const samlArtifact = SAMLart.replace(/ /g, '+')
     const samlArtifactBuffer = Buffer.from(samlArtifact, 'base64')
     const index = samlArtifactBuffer.readInt8(samlArtifactBuffer.length - 1)
+    // use env NRIC as open_login_page = false
+    if (index === -1) index = assertions.identities.singPass.indexOf(singPassNric)
```
Comment (LoneRifle) [inline lib/express/myinfo.js:118]: Would not work given that index is `const`
Nitpick - for clarity, enclose even single-line statements in brackets

**Comment by LoneRifle** [comment]
Thanks for fixing the syntax @yahudev ! Would you be kind enough to also undertake the linting and style changes I have pointed out please?

**Comment by LoneRifle** [comment]
Thanks again @yahudev ! Would you also explain the rationale behind the logic change please?

**Comment by yahudev** [comment]
> Thanks again @yahudev ! Would you also explain the rationale behind the logic change please?

I'm using the mock server for API automated test for myinfo integration. In such case I would like to make use of the SHOW_LOGIN_PAGE flag to set false. From there I would expect the mock service to make use of MOCKPASS_NRIC like it was for soap, as otherwise SHOW_LOGIN_PAGE=false doesn't really work myinfo/authorise endpoint. So I come up with this change to meet my need and expectation. let me know if it doesn't make too much sense.

**Comment by LoneRifle** [comment]
Hello @yahudev - I've recently implemented v3 of the MyInfo API, which presumably you would have to migrate to at some point. Given it's a breaking change (myinfo now has `/v2` and `/v3` endpoints), please be careful when picking up MockPass 2.0.

**Pull request merged**